cluster-operator/src/main/java/io/strimzi/operator/cluster/model/TemplateUtils.java

```diff
@@ -31,6 +42,82 @@ public static Map<String, String> labels(HasMetadataTemplate template)   {
         }
     }
 
+    /**
+     * This method is used to get additional volumes for a given pod template and a list of existing volumes. It validates
+     * the additional volumes for duplicate volume names, conflicting mount paths, and sensitive paths. If the validation
+     * passes, it adds the additional volume to the existing volumes list.
+     *
+     * @param templatePod     The pod template that contains the additional volumes.
+     * @param existingVolumes The list of existing volumes to which the additional volumes will be added.
+     * @return The list of volumes including the additional volumes.
+     * @throws RuntimeException If there is a duplicate volume name.
+     */
+    public static List<Volume> getAdditionalVolumes(PodTemplate templatePod, List<Volume> existingVolumes) {
+        // Extract the names of the existing volumes
+        List<String> existingVolumeNames = existingVolumes.stream().map(Volume::getName).toList();
+
+        // Check if there are any duplicates in the additional volumes' names or paths
+        boolean hasDuplicate = templatePod.getAdditionalVolumes().stream().anyMatch(additionalVolume -> existingVolumeNames.contains(additionalVolume.getName()));
+
+        // Throw an exception if there are any duplicates or sensitive paths
+        if (hasDuplicate) {
+            throw new RuntimeException("Duplicate volume name found in additional volumes");
+        }
+
+        // Add the additional volumes to the existing volumes list
+        templatePod.getAdditionalVolumes().forEach(volumeConfig -> existingVolumes.add(createVolumeFromConfig(volumeConfig)));
+
+        // Return the updated list of volumes
+        return existingVolumes;
+    }
+
+    /**
+     * Add additional volume mounts to the given list of volume mounts. Validation is performed on the additional volume
+     * mounts to ensure the mount paths do not conflict with with the volume mounts in the list to be added to. Validation
+     * is also performed that none of the additional volume mount paths are forbidden.
+     * 
+     * @param volumeMounts           The list of volume mounts to be added to
+     * @param additionalVolumeMounts The list of volume mounts to add
+     * @return The combined list of volume mounts
+     * @throws RuntimeException If there is a conflicting mount path, or a forbidden path.
+     */
+    public static List<VolumeMount> addAdditionalVolumeMounts(List<VolumeMount> volumeMounts, List<VolumeMount> additionalVolumeMounts) {
```

Comment:
Are subpaths and similar being handled here? I have not tested yet.

Reply:
Yes, subpaths etc are being handled (a benefit of using the kubernetes VolumeMount directly).